.github/workflows/tests.yml

```diff
@@ -1,28 +1,57 @@
 name: Run Tests
 
-on: [push, pull_request, workflow_dispatch]
+on: 
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
 
 jobs:
   unittest-and-lint:
-    runs-on: ubuntu-latest
+
+    runs-on: ${{matrix.os}}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ["3.8", "3.10"]
```

Comment:
python 3.11 has been released

Reply:
Fair, I would suggest testing Python 3.8 and the latest version is then sufficient.

Reply:
I agree